### PR TITLE
Automate updating python_releases.txt

### PR DESCRIPTION
**Pull request by CharlieZhao95**
I automate updating `python_releases.txt` based on tags information from cpython repo.
There are some problems compared to manually updating the file:
* The commit date for some tags is different from the actual release date(versions released on [python.org](https://www.python.org/downloads/)), which may differ by 1-2 days;
* Updating the file will take about 1min, so I added a new target to makefile to avoid taking too long, which might be better than integrating the behavior of updating the file in the `make update` target;

### CODE REVIEW

File: render_doc.py [hunk 1]
```diff
@@ -844,6 +847,40 @@ def get_date(self, version):
...
+    def is_release_tag(tag):
+        return all(v not in tag and 'v' in  tag for v in ['a', 'b', 'c', 'rc'])
+
+    @staticmethod
+    def format_version(version):
+        if version.count('.') == 1:
+            version += '.0'
+        return version[1:] if version.startswith('v') else version
+
+    def _get_all_tags(self):
```
Comment (vstinner) [inline render_doc.py:860]: Why not using ``git tag -l`` to list tags and then ``git show TAG`` to retrieve the date?

Comment (CharlieZhao95) [reply render_doc.py:860]: > Why not using `git tag -l` to list tags and then `git show TAG` to retrieve the date?

I'm worried that the workflow will become too long. We first need to know that a new version is released, then we should update the local repo, and finally execute the `make update` command. If you use the API you don't need to know the details of the whole process, just simply execute the command. 🤔 

Of course, using `git tag -l` seems to be a faster and more stable solution to get tag information. 
Do I need to modify the current method of getting tag information?

Comment (CharlieZhao95) [reply render_doc.py:860]: Oh. It seems that the `CommitDates` class has used the commit information from the local repository, I missed it. 
It is better to use `git show TAG` to get date information, I will modify the method to get the date.

**Comment by vstinner** [comment]
Thanks, that's a great enhancement! I made commit 5f1be944bf3d118eef91d70458d1820879d693ce to adjust your code. The main change is about making sure that the doc can be built without having to clone the Python Git repository ("offline" mode by default): only ``make update`` gets tags.

**Pull request merged**